config/confighttp/confighttp.go

```diff
@@ -232,6 +233,7 @@ func (hss *HTTPServerSettings) ToListener() (net.Listener, error) {
 		if err != nil {
 			return nil, err
 		}
+		tlsCfg.NextProtos = []string{http2.NextProtoTLS}
```

Comment:
Yeah the runtime transparently falls back to http/1.1 if http2 is added as of

https://go-review.googlesource.com/c/net/+/325611/

I'm not sure if it's more idiomatic to add http1 explicitly here, I didn't since there isn't a constant for it like there was for http2. Let me know if you have a preference.

Reply:
That changeset is interstiThe associated issue (golang/go/issues/46310) makes me think we should indeed add HTTP/1 to the list, even if there is no constant:

>  there is a risk that servers configured a partial NextProtos list, expecting other protocols to just fallback to not negotiating ALPN.
> 
> In particular, we have anecdotal evidence of servers with NextProtos: []string{"h2"} that expect a client that tries to negotiate http/1.1 to fallback successfully. These servers will break in Go 1.17.

Reply:
> I wonder if this is a typo for Go 1.16?

I am not sure 🤔 the associated changeset was merged for go1.17beta1, so maybe this issue was never a problem on a stable Go version, only on master

> if adding the fallback explicitly is clearer than definitely no problem, please just provide one more nudge that it's preferred.

I would feel more comfortable adding the fallback, yes. I think in practical terms it doesn't matter and it will work either way, but I think adding it explicitly is clearer and consistent with the implementation of `golang.org/net/x/http2`.

> I would need to construct literals (I think it is http/1.1 but http http1 h1 all seem reasonable) until I got the right one for the unit test to pass.

Based on https://cs.opensource.google/go/x/net/+/de3da570:http2/transport.go;l=208-210 I would say the correct string to add is `"http/1.1"`. This also matches the string on [the ALPN RFC](https://datatracker.ietf.org/doc/html/rfc7301#section-6).

Reply:
Cool, I added the fallback